trikot-foundation/test-utils/src/commonMain/kotlin/com/mirego/trikot/foundation/timers/ConfigurableMockTimerFactory.kt

```diff
@@ -0,0 +1,71 @@
+import com.mirego.trikot.foundation.timers.Timer
+import com.mirego.trikot.foundation.timers.TimerFactory
+import kotlin.test.assertTrue
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+data class TimerData(val timer: MockTimer, val fireOffTime: Duration, val repeatTime: Duration = fireOffTime, val repeatable: Boolean = false) {
+    var fired = false
+}
+
+class ConfigurableMockTimerFactory(initialTime: Duration = 0.toDuration(DurationUnit.MILLISECONDS)) : TimerFactory {
+    var singleCall = 0
+    var repeatableCall = 0
+    val timers = mutableListOf<TimerData>()
+    private var currentTime = initialTime
+
+    fun addToTime(timeToAdd: Duration) = setTime(currentTime + timeToAdd)
```

Comment:
I feel like naming this function to simply `add` would make it cleaner and ironically easier to understand!

Reply:
would you rename `setTime` to `set` as well?

Reply:
Actually, naming the methods `addTime` and `setTime` would be a lot better 🙂